Build failure: trayer

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Attempt to build `trayer`and due to GCC 15 being used now, it fails to build due to function declarations without parameters

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
> panel.c: In function 'panel_start_gui':
       > panel.c:315:5: error: too many arguments to function 'bg_init'; expected 0, have 1
       >   315 |     bg_init(gdk_helper_display());
       >       |     ^~~~~~~ ~~~~~~~~~~~~~~~~~~~~
       > In file included from panel.c:15:
       > bg.h:12:6: note: declared here
       >    12 | void bg_init();
```

### Additional context

I have a PR ready, creating an issue just for tracking

### System metadata

n/a

### Notify maintainers

@pSub 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/tr/trayer/package.nix
+++ b/pkgs/by-name/tr/trayer/package.nix
@@ -5,6 +5,7 @@
   pkg-config,
   gdk-pixbuf,
   gtk2,
+  fetchpatch,
 }:
 
 stdenv.mkDerivation rec {
@@ -21,6 +22,15 @@ stdenv.mkDerivation rec {
   postPatch = ''
     patchShebangs configure
   '';
+
+  patches = [
+    # Adding missing arg in function decleration
+    (fetchpatch {
+      name = "fix_function_dec.patch";
+      url = "https://gitweb.gentoo.org/repo/gentoo.git/plain/x11-misc/trayer-srg/files/trayer-srg-1.1.8-fix-define.patch?id=94ae89d1b044c24138d5c8903df68e9654a5462f";
+      hash = "sha256-LighVaBDePheBO+dWG6JHhm/Y6sxdtvTrBar8VrPRH4=";
+    })
+  ];
 
   nativeBuildInputs = [ pkg-config ];
 

